### ISSUE TITLE
Godot Editor stutters randomly

### ISSUE THREAD

**Issue by Yni-Viar** [godotengine/godot#95575]
### Tested versions

- Reproducible in 4.3 after 714effd
- Not reproducible in 4.3 before 714effd , 4.2.2

### System information

Godot v4.3.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1050 Ti (NVIDIA; 32.0.15.6081) - Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz (12 Threads)

### Issue description

I got upgraded my projects to Godot 4.3 from 4.2.2 and noticed, that it stutters randomly (I tested 4.3 Dev 6 and 4.2.2 stable - no issues) if V-Sync is enabled.
Godot 4.3 Beta 2, RC 2 and 3 partially trigger this bug.
Below I posted --verbose log, what Godot did when it was stuttering
```
EditorTheme: Generating new theme for the config '66886095'.
EditorTheme: Can keep old icons, copying.
EditorTheme: Generating new fonts.
EditorTheme: Generating new styles.
```
![изображение](https://github.com/user-attachments/assets/2963476a-84d6-4851-9ed4-92d69f17812d)

### Steps to reproduce

For 4.3.stable: Create empty project -> open the editor -> see stutter randomly
For 4.3 Beta 1, Beta 2, RC 2 and 3: Create empty project  -> open the editor -> set V-Sync to enabled/adaptive in editor settings -> zoom in and out repeatedly (when window is inactive) -> see stutter randomly

### Minimal reproduction project (MRP)

Mostly unneeded, occurs even in empty project.
Here I leave the example one, created in 4.3.stable
[test.zip](https://github.com/user-attachments/files/16627611/test.zip)

**Comment by Yni-Viar** [comment]
Edit: in large projects editor stutters even if V-Sync disabled.

**Comment by Yni-Viar** [comment]
The bug likely was introduced between 4.3 dev 6 and beta 1.

**Comment by Yni-Viar** [comment]
I investigated into the problem, and found, that #89033 is the source of bug. I debugged stable version, and found source of lag. Also I checked build before commit (no issues) and after commit (the bug is triggered).
![2024-08-19 125237](https://github.com/user-attachments/assets/f1e7e324-ef0e-4548-8a50-24636e210751)
![2024-08-19 124740](https://github.com/user-attachments/assets/6589ca34-ae28-4d96-81aa-63909a6dac52)
